File: Day 23 : BST Level-Order Traversal/Solution.py
class Node:
    def __init__(self,data):
        self.right=self.left=None
        self.data = data

class Solution:
    def insert(self,root,data):
        if root==None:
            return Node(data)
        else:
            if data<=root.data:
                cur=self.insert(root.left,data)
                root.left=cur
            else:
                cur=self.insert(root.right,data)
                root.right=cur
        return root
    
    def levelOrder(self, root):
        queue = [root]
        
        while len(queue) != 0:
            front = queue[0]
            queue.pop(0)
            print(front.data, end = ' ')
            if front.left != None:
                queue.append(front.left)
            if front.right != None:
                queue.append(front.right)

if __name__ == "__main__" :
    T=int(input())
    myTree=Solution()
    root=None
    for i in range(T):
        data=int(input())
        root=myTree.insert(root,data)
    myTree.levelOrder(root)

File: Day 23 : BST Level-Order Traversal/Solution.rb
class Node
  attr_accessor :left,:right,:data
  def initialize data
    @left=@right=nil
    @data=data
  end
end

class Solution
  def insert (root,data)
    if root==nil
      return Node.new(data)
    else
      if data<=root.data
          cur=self.insert(root.left,data)
          root.left=cur
      else
          cur=self.insert(root.right,data)
          root.right=cur
      end
    end
    return root
  end

  def levelOrder(root)
    queue = [root]
    while queue.length > 0
      ele = queue.shift
      print "#{ele.data} "
      
      if ele.left != nil
          queue.push ele.left
      end
      if ele.right != nil
          queue.push ele.right
      end
    end
  end
end

myTree = Solution.new
root = nil
T = gets.to_i
for i in 1..T
  data = gets.to_i
  root = myTree.insert(root,data)
end
myTree.levelOrder(root)
